### PR TITLE
KOGITO-4310 - JIT DMN accepts base64 encoded dmn models

### PR DESCRIPTION
**Pull request by r00ta**
JIRA: KOGITO-4310

JIRA: https://issues.redhat.com/browse/KOGITO-4310

This PR aims to accept base64 encoded DMN models on the jit endpoints.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Review by tarilabs — APPROVED** [review]
thank you :+1

File: jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/SchemaResource.java [hunk 1]
```diff
@@ -58,7 +59,7 @@ private static DMNModel modelFromXML(String modelXML) {
     @Consumes(MediaType.APPLICATION_XML)
     @Produces(MediaType.APPLICATION_JSON)
     public Response schema(String payload) {
-        DMNModel dmnModel = modelFromXML(payload);
+        DMNModel dmnModel = modelFromXML(Base64Utils.decode(payload));
```
Comment (tarilabs) [inline jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/SchemaResource.java]: sorry for "changing my mind" but I noticed this only now.
Why do we need to base64 a normal REST payload of type `application/xml` ?

File: jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/utils/Base64Utils.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+public class Base64Utils {
+
+    public static String encode(byte[] plain) {
+        return Base64.getEncoder().encodeToString(plain);
+    }
+
+    public static String decode(String encoded) {
+        byte[] b = Base64.getDecoder().decode(encoded);
+        return new String(b);
+    }
```
Comment (tarilabs) [inline jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/utils/Base64Utils.java:30]:
```suggestion
    }
    
    private Base64Utils() {
        // not allowed for util class.
    }
```

**Review by tarilabs — CHANGES_REQUESTED** [review]

File: jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/utils/Base64Utils.java (hunk 2 above)
Comment (tarilabs) [reply jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/utils/Base64Utils.java:30]: this should avoid SonarCloud complaints 👍

File: jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/SchemaResource.java (hunk 1 above)
Comment (r00ta) [reply jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/SchemaResource.java]: Thanks for the catch @tarilabs ! Fixed

Comment (tarilabs) [reply jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/SchemaResource.java]: This was a REST endpoint taking XML.
Imho, it should still be a REST for XML input, JSON output.
REST is simple and I would keep it as it was previous this PR.

Comment (r00ta) [reply jitexecutor/jitexecutor-dmn/src/main/java/org/kie/kogito/jitexecutor/dmn/api/SchemaResource.java]: Ok, reverted

**Comment by r00ta** [comment]
Not needed anymore as discussed on gchat

**Pull request closed without merging**